cmd/client-s3.go

```diff
@@ -2089,6 +2089,19 @@ func (c *s3Client) PutObjectRetention(mode *minio.RetentionMode, retainUntilDate
 	return nil
 }
 
+// Set object legal hold for a given object.
+func (c *s3Client) PutObjectLegalHold(lhold *minio.LegalHoldStatus) *probe.Error {
+	bucket, object := c.url2BucketAndObject()
```

Comment:
should we check for either bucketname or objectname being empty?

Reply:
it is already being done in minio-go, it is not necessary to duplicate this check here